velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -2281,3 +2281,97 @@ TEST_F(StringFunctionsTest, concatInSwitchExpr) {
       {"This is a long sentence-zzz"_sv, "aaa-This is some other sentence"_sv});
   test::assertEqualVectors(expected, result);
 }
+
+TEST_F(StringFunctionsTest, fromBigEndian32) {
+  const auto fromBigEndian32 = [&](std::optional<std::string> arg) {
```

Comment:
Please add some corner test cases say where we try to convert a very large value (string > 12 char, all F chars 'FFFFFFFFFFFFFFFFFFFFFFFF') and validate the behavior.